Fix freeProxyCalls and add comment in list.c

ChungHsuanChen opened the pull request:
In test.c ,when list_destroy greater than 2 times and doing assert , it won't have correct freeProxyCalls.

![Test_bug](https://user-images.githubusercontent.com/73817114/147570639-6101c54d-cc93-446d-a151-d2be42cb11cf.png)

Therefore, every time initialize freeProxyCalls is essential.
Add Comment about parameter in list.c.

stephenmathieson commented on src/list.c:
```diff
@@ -26,6 +26,7 @@ list_new(void) {
 
 /*
  * Free the list.
+   @self: Pointer to the list 
```
Please keep comment style consistent.

```suggestion
 * @self: Pointer to the list 
```

The rest of the comments should be updated to match the above.

ChungHsuanChen replied:
Finish Refinement, And I also commited it.

stephenmathieson reviewed (changes requested):
Code style is off.

Also, I'm not sure what this is fixing. The changes here only appear to be whitespace and comments.

ChungHsuanChen commented:
OK, I have changed it.
The code I fixed is about if static variable freeProxyCalls aren't initialized after destroyed the list , we will get wrong answer when we continuously create and destroy a list(I have wrote a function named test_list_destroy_complexver()).
I will push again after refining code style.
Thanks.

nouwaarom commented:
@ChungHsuanChen Thanks for improving the tests a bit! Can you please rebase this branch so it can be merged?

ChungHsuanChen commented:
Ok, I have rebased it.

nouwaarom commented:
Thanks for your contribution!
